### ISSUE TITLE
Multiple problems with PyQt5 Qt5Network import / SSL

### ISSUE THREAD

**Issue by gehlhaarPfizer** [pyinstaller/pyinstaller#4048]
Running PyInstaller 3.4, Windows 10, PyQt 5.11.3, Python 3.6.8.

Problem 1: PyInstaller (hook-PyQt5.QtNetwork.py) adds SSL library dependencies even if Qt5 is built without SSL support; it should be loading the Qt5Network library and checking QSslSocket::SupportsSsl() before requiring the SSL libraries be present.

Problem 2: Qt does not install these libraries (libeay32.dll, ssleay32.dll) into the directory where PyInstaller is looking for them; indeed they are completely external to Qt. PyInstaller should be looking for them in the system PATH, not the Qt BinariesPath.

**Comment by bjones1** [comment]
Thanks for the bug report. Would you test the updated version of `hook-PyQt5.QtNetwork.py` to see if this fixes the problem? Also, for problem 2, can you provide a link to the Qt docs which documents this? I'd like to include that in the code.

``` Python
#-----------------------------------------------------------------------------
# Copyright (c) 2013-2019, PyInstaller Development Team.
#
# Distributed under the terms of the GNU General Public License with exception
# for distributing bootloader.
#
# The full license is in the file COPYING.txt, distributed with this software.
#-----------------------------------------------------------------------------
import os.path

from PyInstaller.utils.hooks import add_qt5_dependencies, eval_statement
from PyInstaller.compat import is_win
from PyInstaller.depend.bindepend import getfullnameof

hiddenimports, binaries, datas = add_qt5_dependencies(__file__)

# Add libraries needed for SSL if these are available. See issue #3520, #4048.
if (is_win and eval_statement("""
    from PyQt5.QtNetwork import QSslSocket
    print(QSslSocket.supportsSsl())""")):

    rel_data_path = ['PyQt5', 'Qt', 'bin']
    binaries += [
        (getfullnameof('libeay32.dll'), os.path.join(*rel_data_path)),
        (getfullnameof('ssleay32.dll'), os.path.join(*rel_data_path)),
    ]
```

**Comment by gehlhaarPfizer** [comment]
I will test this and get back to you today. At first glance it looks good.

The relevant Qt documentation is here:
http://doc.qt.io/qt-5/ssl.html

Although it doesn't specifically say that it doesn't "install" those DLLs, it is clear that they are searched for externally, e.g. from this page:

By default, an SSL-enabled Qt library dynamically loads any installed OpenSSL library at run-time. However, it is possible to link against the library at compile-time by configuring Qt with the -openssl-linked option.

When building a version of Qt linked against OpenSSL, the build system will attempt to link with libssl and libcrypt libraries located in the default location on the developer's system. This location is configurable: set the OPENSSL_LIBS environment variable to contain the linker options required to link Qt against the installed library. For example, on a Unix/Linux system:

OPENSSL_LIBS='-L/opt/ssl/lib -lssl -lcrypto' ./configure -openssl-linked

To disable SSL support in a Qt build, configure Qt with the -no-openssl option.

**Comment by bjones1** [comment]
@gehlhaarPfizer, thanks -- the doc link is very helpful. Here's my updated hook. Tweaks/fixes welcome.

``` Python
#-----------------------------------------------------------------------------
# Copyright (c) 2013-2019, PyInstaller Development Team.
#
# Distributed under the terms of the GNU General Public License with exception
# for distributing bootloader.
#
# The full license is in the file COPYING.txt, distributed with this software.
#-----------------------------------------------------------------------------
import os.path

from PyInstaller.utils.hooks import add_qt5_dependencies, eval_statement
from PyInstaller.compat import is_win
from PyInstaller.depend.bindepend import getfullnameof

hiddenimports, binaries, datas = add_qt5_dependencies(__file__)

# Add libraries needed for SSL if these are available. See issue #3520, #4048.
if (is_win and eval_statement("""
    from PyQt5.QtNetwork import QSslSocket
    print(QSslSocket.supportsSsl())""")):

    rel_data_path = ['PyQt5', 'Qt', 'bin']
    binaries += [
        # Per http://doc.qt.io/qt-5/ssl.html#enabling-and-disabling-ssl-support,
        # the SSL libraries are dynamically loaded, implying they exist in
        # the system path. Include these.
        (getfullnameof('libeay32.dll'), os.path.join(*rel_data_path)),
        (getfullnameof('ssleay32.dll'), os.path.join(*rel_data_path)),
    ]
```

**Comment by gehlhaarPfizer** [comment]
Tested -- seems to work.

**Comment by bjones1** [comment]
Thanks -- this is now merged as 43774a033bc096f8e46a0c4948214eeb5c7d979c.
